src/core/analysis/stoppingRule/GelmanRubinStoppingRule.cpp

```diff
@@ -100,13 +98,34 @@ bool GelmanRubinStoppingRule::stop( size_t g )
         }
         
         // set the burnins
-        for ( size_t j = 0; j < data.size(); ++j)
+        for ( size_t j = 0; j < data[i].size(); ++j)
+        {
+            data[i][j].setBurnin(maxBurnin);
+        }
+        
+        // get the number of variables in the current replicate
+        num_variables_in_rep = data[i].size();
+        
+        // if we have different numbers of variables in different replicates, something has gone terribly wrong
+        if (i > 1 && data[i].size() != num_variables_in_rep)
+        {
+            throw RbException("The replicates contain different sets of parameters.");
+        }
+    }
+    
+    // invert the hierarchy
+    std::vector< std::vector<TraceNumeric> > data_exp;
+    data_exp.reserve( num_variables_in_rep );
+    
+    for (size_t i = 0; i < num_variables_in_rep; i++)
+    {
+        for ( size_t j = 0; j < numReplicates; j++)
         {
-            data[j].setBurnin(maxBurnin);
+            data_exp[i].push_back( data[j][i] );
         }
         
         // conduct the test
-        passed &= grTest.assessConvergence(data);
+        passed &= grTest.assessConvergence(data_exp[i]);
```

Comment:
Why not just return `false` here if `!grTest.assessConvergence(data_exp[i])`?

Reply:
Hmm, that's a good point. This solution (which was already in the code; I just left it unchanged) is more elegant, since we return the test itself rather than _conditional on_ the test, but it could lead to a lot of unnecessary computation – e.g., if you have a complex model with several hundred parameters and the 1st or 2nd one already fails the test. I'll change it.